### PR TITLE
Interp freq first

### PR DESCRIPTION
**Pull request by steven-murray**
Faster performance when using `beam_list` and the `freq` is not in the `beam.freq_array`.

### CODE REVIEW

File: src/vis_cpu/vis_cpu.py [hunk 1]
```diff
@@ -194,6 +194,15 @@ def vis_cpu(
             0 <= i < nbeam for i in beam_idx
         ), "beam_idx contains indices greater than the number of beams"
 
+    if beam_list is not None:
+        # make sure we interpolate to the right frequency first.
+        beam_list = [
+            b.interp(freq_array=np.array([freq]), new_object=True, run_check=False)
+            if (isinstance(freq, UVBeam) and freq not in b.freq_array)
```
Comment (r-pascua) [inline src/vis_cpu/vis_cpu.py:201]: I think you mean `isinstance(b, UVBeam)` here.

**Review by r-pascua — COMMENTED** [review]
Looks fine to me, just need to address my comment before merging (otherwise this won't work). One other comment is that I'd prefer changing `b` to `bm` or `beam`, but I don't think it's essential to make that change.

One thing to consider, though, is that the `in` operation is not flexible with floating point errors. Suppose `freq` is like 1 Hz away from a frequency in the beam's `freq_array` and we have a channel width of ~100 kHz&mdash;do we really want to interpolate in this case? (This is a bit bigger than a floating point error, but you get the idea.)

On another note, I think we want to remove `freq_array=...` from line 300, since the interpolation is already done before the time loop is entered.

File: src/vis_cpu/vis_cpu.py (hunk 1 above)
Comment (steven-murray) [reply src/vis_cpu/vis_cpu.py]: Yup.

**Comment by steven-murray** [comment]
Thanks @r-pascua, that was a useful review.

So, I ended up ALWAYS interpolating to `freq` if its a UVBeam -- I think that operation is short-circuited properly inside of `interp` if the frequency is in the data array already (taking into account a tolerance). Plus, that was the only way to ensure that I could safely remove the `freq_array` on line 300 (otherwise some of the beams would interpolate at all native frequencies).

**Comment by r-pascua** [comment]
Ah, okay, that makes sense. It looks like my suggestion of removing the specification of `freq_array` in the call to `interp` on line 300 was bunk, since `UVBeam.interp` apparently requires `freq_array`. I had assumed this was optional&mdash;sorry for suggesting a buggy change.

**Review by r-pascua — APPROVED** [review]
OK, I think this all looks good! Merge whenever you're ready.

**Pull request merged**